### PR TITLE
[11.0][IMP] maintenance_plan

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
Mian improvments included in this PR:

* Extension of maintenance plan module with the addition and renaming of the following fields:
    - Start date
    - Planning Horizon & Plannig Horizon step
    - Instructions page
    - period --> interval
    - Interval setp
* Creation of maintenance requests taking into account planning horizon

Pending:
* [x] Add migration scripts

cc ~ @Eficent

### CODE REVIEW

**Comment by AdriaGForgeFlow** [comment]
ping ~ @grindtildeath

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
This:
image

Caused:

image

**Comment by AdriaGForgeFlow** [comment]
> This:
> image
> 
> Caused:
> 
> image

Could you explain the steps to reproduce the error? It worked for me:

maintenance_plan

**Comment by AaronHForgeFlow** [comment]
Yeap I got that error on runbot on Friday. But after rebuilding runbot this morning I am not able to reproduce, so probably the problem was only on that runbot instance.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Tested on runbot

File: maintenance_plan/models/maintenance_equipment.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# Copyright 2017 Camptocamp SA
+# Copyright 2019 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from dateutil.relativedelta import relativedelta
+from odoo import _, api, fields, models
+
+
+def get_relativedelta(interval, step):
```
Comment (grindtildeath) [inline maintenance_plan/models/maintenance_equipment.py]: As this is already defined in `maintenance_plan.py` you could just import it with `from .maintenance_plan import get_relativedelta`

File: maintenance_plan/views/maintenance_kind_views.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        <field name="name">Maintenance kinds</field>
+        <field name="res_model">maintenance.kind</field>
+        <field name="view_mode">tree</field>
+        <field name="view_id" ref="maintenance_kind_view_tree"/>
+    </record>
+
+    <menuitem
+        id="menu_maintenance_kind_configuration"
+        name="Maintenance kinds"
+        parent="maintenance.menu_maintenance_configuration"
```
Comment (grindtildeath) [inline maintenance_plan/views/maintenance_kind_views.xml:25]: I don't see this menu on runbot, but don't have time to look into it.

**Review by grindtildeath — COMMENTED** [review]
Looks like a good improvement overall, thanks :+1: 

Just tested on runbot, and seems to work well out of the missing menu.

BTW, shouldn't you add migration script for existing records to be migrated with the new fields ?

File: maintenance_plan/models/maintenance_equipment.py (hunk 1 above)
Comment (JordiBForgeFlow) [reply maintenance_plan/models/maintenance_equipment.py]: Thanks! Fixed.

File: maintenance_plan/views/maintenance_equipment_views.xml [hunk 3]
```diff
@@ -15,7 +15,7 @@
                     <field string="Plans" name="maintenance_plan_count" widget="statinfo"/>
                 </button>
             </div>
-            <xpath expr="//group[@name='maintenance']" position="replace"/>
+            <xpath expr="//notebook/page[3]" position="replace"/>
```
Comment (LoisRForgeFlow) [inline maintenance_plan/views/maintenance_equipment_views.xml]: Why are you hiding this?

**Comment by JordiBForgeFlow** [comment]
@bestmazzo @astirpe could you please take a few minutes to review this one? It changes the workflow.

**Comment by dalonsod** [comment]
Hello,
Could you take a look at #33 and #35 ? I think these issues are still unfixed in v11 in this module, maybe this PR is a good place to add those changes.

**Comment by AdriaGForgeFlow** [comment]
> Hello,
> Could you take a look at #33 and #35 ? I think these issues are still unfixed in v11 in this module, maybe this PR is a good place to add those changes.

For #33, in v11 the maintenance request model does not have a company_id field. We could discuss about adding this field. @jbeficent  What do you think?
Regarding #35 I have added the improvement in the last commit.

Thanks for your comments!

File: maintenance_plan/views/maintenance_equipment_views.xml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        <field name="arch" type="xml">
+            <div name="button_box" position="inside">
+                <button name="%(maintenance_plan_equipment_action)d"
+                        type="action"
+                        class="oe_stat_button"
+                        icon="fa-list">
+                    <field string="Plans" name="maintenance_plan_count" widget="statinfo"/>
+                </button>
+            </div>
+            <xpath expr="//group[@name='maintenance']" position="replace">
```
Comment (jarroyomorales) [inline maintenance_plan/views/maintenance_equipment_views.xml]: I think it is better to replace the group above it so the one2many list can be expanded to the whole screen.
//group[@name='maintenance']/..

imagen

**Review by jarroyomorales — APPROVED** [review]
LGTM :+1: , just left a minor comment

File: maintenance_plan/views/maintenance_equipment_views.xml (hunk 4 above)
Comment (AdriaGForgeFlow) [reply maintenance_plan/views/maintenance_equipment_views.xml]: I did not know about this notation. Thanks! :+1:

**Review by etobella — APPROVED** [review]
LGTM :+1:

**Review by JordiBForgeFlow — APPROVED** [review]

**Pull request merged**